captain/models/topology.py

```diff
@@ -117,11 +122,14 @@ async def run_job(self, job_id: str):
                 previous_jobs=previous_jobs,
             )
         )
+        if node["cmd"] == "LOOP":
+            self.loop_nodes.append(job_id)
```

Comment:
There is an opportunity here to make the code more readable, maybe have a function is_loop_node(node_id) then put the node = cast(... ad the if condition there, makes for better code :) your call

Reply:
Adjusted as per your recommendation, thanks